src/java.base/share/classes/sun/invoke/util/VerifyAccess.java

```diff
@@ -284,12 +284,14 @@ public static boolean isTypeVisible(Class<?> type, Class<?> refc) {
         if (refcLoader == null && typeLoader != null) {
             return false;
         }
-        if (typeLoader == null && type.getName().startsWith("java.")) {
-            // Note:  The API for actually loading classes, ClassLoader.defineClass,
-            // guarantees that classes with names beginning "java." cannot be aliased,
-            // because class loaders cannot load them directly.
-            return true;
-        }
+
+        // The API for actually loading classes, ClassLoader.defineClass,
+        // guarantees that classes with names beginning "java." cannot be aliased,
+        // because class loaders cannot load them directly. However, it is beneficial
+        // for JIT-compilers to ensure all signature classes are loaded.
```

Comment:
Since we anticipate this method to perform side effects, can we rename all of these non-pure `isTypeVisible` to `seeType`/`accessType` to indicate this desired side effect?